### PR TITLE
Add waterbodies to framework base functions

### PR DESCRIPTION
**Pull request by jameshalgren**
Water bodies are now parsed and returned in the 'supernetwork values' collection returned by the nhd_network_utilities function. These are not used further by this version of code but a later version will use the waterbodies to add further discretization to the network traversal.

### CODE REVIEW

**Comment by jameshalgren** [comment]
Closes #5

**Comment by jameshalgren** [comment]
Juzer,
Is it correct that the change to WPC affects only the qdc and not the velc
and depthc?
--
James Halgren
NOAA National Water Center
+1 205-347-1453
Lynker <https://www.lynker.com/>

File: src/python_framework/networkbuilder.py [hunk 1]
```diff
@@ -33,9 +32,58 @@ def get_down_connections(
...
+    , data_key = r'data'
+    , downstream_key = r'downstream'
+    , upstreams_key = r'upstreams'
+    , verbose = False
+    , debuglevel = 0
+    ):
+
+    if verbose: print('waterbody_set ...')
+    waterbody_set = {con[data_key][waterbody_col] for key, con in connections.items()}
+    waterbody_set.remove(waterbody_null_code)
```
Comment (groutr) [inline src/python_framework/networkbuilder.py:49]: this should probably use `set.discard`, unless you want an error thrown if the null code doesn't exist.

**Comment by jameshalgren** [comment]
@jdhondia-noaa I just recommitted with the full correction to WPC using proper declaration and passing in the fortran code so that WPC is not used uninitialized.

File: src/python_framework/networkbuilder.py (hunk 1 above)
Comment (jameshalgren) [reply src/python_framework/networkbuilder.py]: Thanks -- made that change.

**Pull request merged**